tck/src/main/java/org/eclipse/microprofile/lra/tck/LRAClientOps.java

```diff
@@ -0,0 +1,196 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck;
+
+import org.eclipse.microprofile.lra.annotation.LRAStatus;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+import org.eclipse.microprofile.lra.client.GenericLRAException;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.temporal.ChronoUnit;
+
+import static junit.framework.TestCase.fail;
+import static org.eclipse.microprofile.lra.tck.participant.api.NonParticipatingTckResource.END_PATH;
+import static org.eclipse.microprofile.lra.tck.participant.api.NonParticipatingTckResource.START_BUT_DONT_END_PATH;
+import static org.eclipse.microprofile.lra.tck.participant.api.NonParticipatingTckResource.STAUS_CODE_QUERY_NAME;
+import static org.eclipse.microprofile.lra.tck.participant.api.NonParticipatingTckResource.TCK_NON_PARTICIPANT_RESOURCE_PATH;
+import static org.eclipse.microprofile.lra.tck.participant.api.ParticipatingTckResource.JOIN_WITH_EXISTNG_LRA_PATH;
+import static org.eclipse.microprofile.lra.tck.participant.api.ParticipatingTckResource.LEAVE_PATH;
+import static org.eclipse.microprofile.lra.tck.participant.api.ParticipatingTckResource.TCK_PARTICIPANT_RESOURCE_PATH;
+
+public class LRAClientOps implements Cloneable {
+    /**
+     * TODO recovery needs to be properly speced
+     * It is used in the two tests for participants that return 202:
+     * {@link TckTests#acceptCancelTest()}  and {@link TckTests#acceptCloseTest()}
+     * to connect to an endpoint that will trigger recovery (the alternative would
+     * be to wait)
+     */
+
+    /**
+     * Key for looking up the config property that specifies which JAX-RS path a
+     * recovery coordinator is running on
+     */
+    static final String LRA_RECOVERY_HOST_KEY = "lra.http.recovery.host";
+
+    /**
+     * Key for looking up the config property that specifies which JAX-RS path a
+     * recovery coordinator is running on
+     */
+    static final String LRA_RECOVERY_PORT_KEY = "lra.http.recovery.port";
+
+    /**
+     * Key for looking up the config property th
+     * recovery coordinator is running on
+     */
+    static final String LRA_RECOVERY_PATH_KEY = "lra.coordinator.recovery.path";
+
+    private final WebTarget target;
+
+    public LRAClientOps(WebTarget target) {
+        this.target = target;
+    }
+
+    private int getStatus(String lra) {
+        // call a JAX-RS endpint that should result in the enlistment of a resource into the LRA
+        int status = invokeRestEndpointAndReturnStatus(lra, TCK_PARTICIPANT_RESOURCE_PATH, JOIN_WITH_EXISTNG_LRA_PATH, 200);
```

Comment:
This assumes the coordinator operates over JAX-RS which is not required. This `LRAClientOps` needs to be an interface and an implementation specific class needs to be provider for the TCK.

Reply:
My bad, nothing coordinator specific here. Ignore my comment.